src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,144 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
+@SuppressWarnings("unchecked")
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int size;
+    private Object[] elementData;
+
+    public ArrayList() {
+        elementData = new Object[DEFAULT_CAPACITY];
+    }
+
+    public void rangeCheck() {
+        if (size == elementData.length) {
+            grow();
+        }
+    }
+
+    public void indexCheck(int index) {
+        if (index >= size) {
+            throw new ArrayListIndexOutOfBoundsException("Invalid index");
+        }
+
+        if (index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Invalid index");
+        }
+    }
+
+    private Object[] grow() {
+        int newCapacity = elementData.length * 3 / 2 + 1; // Новий розмір (збільшення на 50%)
+        elementData = Arrays.copyOf(elementData, newCapacity); // Копіюємо елементи
+        return elementData;
+    }
+
+    public void checkIfElementExsist(T element) {
```

Comment:
There is a typo in the method name `checkIfElementExsist`. It should be `checkIfElementExists` to reflect correct spelling.